src/parser.y

```diff
@@ -450,6 +454,18 @@ unary
             }
 		;
 
+exponent
+       : call
+       | call TOK_EXPONENT exponent
+           {
+              $$ = new BinaryOp($1, BinaryOp::Op::Exponent, $3, LOCD("exponent", @$));
+           }
+       | call TOK_EXPONENT '-' exponent
+          {
+             $$ = new BinaryOp($1, BinaryOp::Op::Exponent, new UnaryOp(UnaryOp::Op::Negate, $4, LOCD("negate", @$)), LOCD("exponent", @$));
+          }
+       ;
```

Comment:
I need to do some more testing, but the following seems to work without any special cases and agrees with the results producted by `python3` for everything I've tried so far.
```
exponent
       : call
       | call TOK_EXPONENT unary
```
Looking at this [python grammar](https://docs.python.org/3/reference/grammar.html) it also matches the `power` declaration.
```
factor: ('+'|'-'|'~') factor | power
power: atom_expr ['**' factor]
atom_expr: [AWAIT] atom trailer*
```
This way we can have any `unary` in the expression.
```
$ python3
>>> 4 ** -+- 3 ** 2
262144
>>> 4 ** +-+ 3 ** 2
3.814697265625e-06
```
Lower priority expressions are resolved without problems too.
```
$ python3
>>> 4**3*3**2
576
>>> 4**3+3**2
73
```
Technically this allows also the unary `!`, I'm not sure if it would make sense trying to prevent that.

Reply:
I have implemented the solution you suggested above in the latest commit, haven't had the chance to see what ! would do but all the test pass.

Reply:
Thanks, for all tests I can currently think of, we get the same result as with `python` which is probably a good reference.